test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -44,6 +44,35 @@ protected override void GetLibrariesToPreload(List<string> libraries)
         public void FixtureSetup()
         {
             ExecutionEvents.GraphPostExecution += ExecutionEvents_GraphPostExecution;
+            
+            //Clear Temp directory folders before start of the new serialization test run
+            var tempPath = Path.GetTempPath();
+            var jsonFolder = Path.Combine(tempPath, "json");
+            var jsonNonGuidFolder = Path.Combine(tempPath, "jsonNonGuid");
+
+            var jsonFilesList = new List<string>();
+            var jsonNonGuidFilesList = new List<string>();
+
+            if (Directory.Exists(jsonFolder) || Directory.Exists(jsonNonGuidFolder))
+            {
+
+                //Try and delete all the files from the previous run. 
+                //If there's an error in deleting files, the tests should countinue
+
+                try
+                {
+                    jsonFilesList = Directory.GetFiles(jsonNonGuidFolder, "*.*", SearchOption.AllDirectories).ToList();
```

Comment:
if `jsonFolder` exists but `jsonNonGuidFolder` does not, this will prevent deletion of the `jsonFolder` files.  These should probably be in separate try/catch blocks.
Can you use `Directory.Delete(jsonNonGuidFolder, true)` here instead of deleting the files individually?

Reply:
agree with @gregmarr 👍

Reply:
The current "delete files individually" method has the problem that as soon as you try to delete a file for which you don't have delete permission (or it's currently locked), then it aborts the entire operation.

Reply:
added individual `try...catch` for each folder.